lib/typing/decimal/decimal.go

```diff
@@ -52,17 +52,15 @@ func (d *Decimal) Precision() *int {
 // This is particularly useful for Snowflake because we're writing all the values as STRINGS into TSV format.
 // This function guarantees backwards compatibility.
 func (d *Decimal) String() string {
-	return d.value.Text('f', d.scale)
-}
-
-func (d *Decimal) Value() any {
-	// -1 precision is used for variable scaled decimal
-	// We are opting to emit this as a STRING because the value is technically unbounded (can get to ~1 GB).
-	if d.precision != nil && (*d.precision > MaxPrecisionBeforeString || *d.precision == -1) {
-		return d.String()
+	targetExponent := -int32(d.scale)
+	value := d.value
+	if value.Exponent != targetExponent {
+		value = DecimalWithNewExponent(value, targetExponent)
 	}
+	return value.Text('f')
+}
 
-	// Depending on the precision, we will want to convert value to STRING or keep as a FLOAT.
+func (d *Decimal) Value() *apd.Decimal {
```

Comment:
We were only calling `.Value()` in one place so I changed it to call `.String()` instead.